tests/lang_callable/macros/tmacros_various.nim

```diff
@@ -346,3 +347,22 @@ block:
   static:
     echo treeRepr(C1[1])
     echo treeRepr(C2[1])
+
+block:
+  # Ensure nkType equality is not ignored by `==` for NimNode
+  macro checkEq(a, b: typed) =
+    echo a == b
+
+  type Exception1 = object of Exception
+  type Exception2 = object of Exception
+  checkEq (;
+    try:
+      discard
+    except Exception1:
```

Comment:
Shouldn't identifier/symbol equality check cause these to be unequal, without needing to consider type?

I think the only way to test this would be macro cache and having two object types with otherwise the same name.

Reply:
These get turned into nkType during sem so there isn't any identifier or sym left in the non-type ast